serde_derive/src/lib.rs

```diff
@@ -71,7 +72,7 @@ pub fn derive_serialize(input: TokenStream) -> TokenStream {
     let input: DeriveInput = syn::parse(input).unwrap();
     match ser::expand_derive_serialize(&input) {
         Ok(expanded) => expanded.into(),
-        Err(msg) => panic!(msg),
+        Err(msg) => TokenStream::from_str(&format!("compile_error!({:?});", msg)).unwrap(),
```

Comment:
Typically macros try to avoid working with Rust source code as strings -- strings are unfriendly to our text editor or IDE when it comes to brace matching, syntax highlighting, indentation, etc. Serde's derive currently uses the [`quote`](https://github.com/dtolnay/quote) crate instead of strings of source code.

Here it's just one line so there may not be an appreciable advantage, but I think it would be better to be consistent with the code generation done in the rest of serde_derive and use `quote!` here as well.